backstory/src/test/java/com/google/sps/vision/VisionManagerImplTest.java

```diff
@@ -0,0 +1,159 @@
+import java.io.IOException;
+import com.google.cloud.vision.v1.AnnotateImageResponse;
+import com.google.cloud.vision.v1.AnnotateImageRequest;
+import com.google.cloud.vision.v1.BatchAnnotateImagesResponse;
+import com.google.cloud.vision.v1.EntityAnnotation;
+import com.google.cloud.vision.v1.Feature;
+import com.google.cloud.vision.v1.Image;
+import com.google.cloud.vision.v1.ImageAnnotatorClient;
+import com.google.protobuf.ByteString;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.Assert;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import org.junit.Test;
+import java.io.ByteArrayOutputStream;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import javax.imageio.ImageIO;
+import java.lang.IllegalArgumentException;
+import com.google.sps.vision.VisionManagerImpl;
+import org.junit.Before;
+
+/**
+ * Tests for VisionManagerImpl.
+ */
+public final class VisionManagerImplTest {
+
+  // Testing strategy for each operation of VisionManagerImpl:
+  /*
+   * public VisionManagerImpl(byte[] rawImageData);
+   *  Partition on rawImageData: empty, null, non-empty image data.
+   * 
+   * public String getLabelsAsJson();
+   *  Unnecesary to test as it is a wrapper for a Gson call.
+   *
+   * public List<String> getLabelDescriptions();
+   *  Partition on this: labelAnnotations is empty, or non-empty list of EntityAnnotations.
+   */
+
+  private byte[] nullRawImageData;
+  private byte[] emptyRawImageData;
+  private byte[] rawImageData;
+  private List<EntityAnnotation> emptyLabelAnnotations;
+  private List<EntityAnnotation> labelAnnotations;
+
+  @Before
+  public void setUp() {
+    nullRawImageData = null;
+    emptyRawImageData = new byte[0];
+    rawImageData = getBytesFromImageReference(
+      "src/test/java/com/google/sps/vision/data/dogRunningOnBeach.jpg", "jpg");
+    emptyLabelAnnotations = new ArrayList<>();
+
+    labelAnnotations = new ArrayList<>();
+
+    EntityAnnotation entityAnnotationMock = mock(EntityAnnotation.class);
+    // mock the behavior of stock service to return the value of various stocks
+    when(entityAnnotationMock.getDescription()).thenReturn("DescriptionOne");
+
+    labelAnnotations.add(entityAnnotationMock);
+    labelAnnotations.add(entityAnnotationMock);
+  }
+
+  /**
+   * Returns an image as a byte array from the local reference of the image.
+   *
+   * @param reference the local reference of the image.
+   * @param fileType the file extension of the image.
+   * @return the byte array representation of the image.
+   */
+  private static byte[] getBytesFromImageReference(String reference, String fileType) {
+    ByteArrayOutputStream bos = new ByteArrayOutputStream();
+
+    try {
+      BufferedImage bImage = ImageIO.read(new File(reference));
+      ImageIO.write(bImage, fileType, bos);
+    } catch (IOException exception){
+      System.err.println("IOException while reading image");
```

Comment:
Can the exception be thrown instead of a printed message?

Reply:
Good idea!

Reply:
Done.